crates/ere-dockerized/src/docker.rs

```diff
@@ -0,0 +1,279 @@
+use std::{
+    fmt::{self, Display, Formatter},
+    fs::{self},
+    io,
+    path::{Path, PathBuf},
+    process::Command,
+};
+use tempfile::TempDir;
+
+#[derive(Clone)]
+pub struct CmdOption(String, Option<String>);
+
+impl CmdOption {
+    pub fn new(key: impl AsRef<str>, value: impl AsRef<str>) -> Self {
+        Self(to_string(key), Some(to_string(value)))
+    }
+
+    pub fn flag(key: impl AsRef<str>) -> Self {
+        Self(to_string(key), None)
+    }
+}
+
+impl Display for CmdOption {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        let Self(key, value) = self;
+        match value {
+            Some(value) => write!(f, "--{key}={value}"),
+            None => write!(f, "--{key}"),
+        }
+    }
+}
+
+#[derive(Clone)]
+pub enum DockerInstruction {
```

Comment:
Instead of building a dockerfile using DockerInstructions, would we have less of a diff, if we had a template Dockerfile and then just "filled in" the parts we wanted?

```rust
ENV {{placeholder}}
```

For example